src/sdp_proxy/advertising_proxy.cpp

```diff
@@ -274,13 +274,30 @@ otbrError AdvertisingProxy::PublishHostAndItsServices(const otSrpServerHost *aHo
 
     if (!hostDeleted)
     {
-        std::vector<uint8_t> firstHostAddress{std::begin(hostAddress[0].mFields.m8),
-                                              std::end(hostAddress[0].mFields.m8)};
+        std::vector<std::vector<uint8_t>> addresses;
+        const otIp6Address *              meshLocalEid = otThreadGetMeshLocalEid(GetInstance());
 
+        addresses.reserve(hostAddressNum);
         // TODO: select a preferred address or advertise all addresses from SRP client.
         otbrLogDebug("Publish SRP host '%s'", fullHostName.c_str());
+
+        for (size_t i = 0; i < hostAddressNum; ++i)
+        {
+            if (otIp6PrefixMatch(meshLocalEid, &hostAddresses[i]) >= OT_IP6_PREFIX_BITSIZE)
+            {
+                otbrLogDebug("Skipping Mesh Local address: %s",
+                             Ip6Address(hostAddresses[i].mFields.m8).ToString().c_str());
+                continue;
+            }
+            addresses.emplace_back(std::begin(hostAddresses[i].mFields.m8), std::end(hostAddresses[i].mFields.m8));
+        }
+        if (addresses.empty())
+        {
+            OnMdnsPublishResult(updateId, error = OTBR_ERROR_NONE);
```

Comment:
Shouldn't an error be returned to fail this SRP registration?

Reply:
No, that's a success.

Reply:
Sorry I didn't get the point, can you provide some background?

> This PR lets advertising proxy filter out mesh local addresses when publishing SRP hosts. If an SRP host only has mesh local addresses, the registration will fail.

The PR description seems indicating that we should fail in this case?

Reply:
@wgtdkp Sorry I forgot to update the PR description. I just had it updated.